### PR TITLE
Issue #12488: 'R: Push Release Notes' makes xml section version as snapshot

### LINKED ISSUE #12488: 'R: Push Release Notes' makes xml section version as snapshot
Detected at https://github.com/checkstyle/checkstyle/issues/12479#issuecomment-1333069436

Execution of action:
https://github.com/checkstyle/checkstyle/actions/runs/3588732543/jobs/6040435644#step:4:1

Results in snapshot version in xml. That is not correct.

Detected:
`<section name="Release 10.5.1-SNAPSHOT">`

Expected:
`<section name="Release 10.5.1">`

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #12488 

Action: https://github.com/stoyanK7/checkstyle/actions/runs/3604870995/jobs/6074684061#step:4:241

### CODE REVIEW

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (rnveach) [inline .ci/releasenotes-gen-xdoc-push.sh:48]: Just curious why this wasn't an issue before? I checked and the last release did not update the POM until after the release notes were committed.

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (romani) [inline .ci/releasenotes-gen-xdoc-push.sh:9]: Fix name of script

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:48]: We never used this script before
We always use
https://github.com/checkstyle/checkstyle/actions/runs/3588930515/jobs/6040835341#step:3:1
`.ci/releasenotes-gen.sh`

Comment (rnveach) [reply .ci/releasenotes-gen-xdoc-push.sh:48]: Then are we fixing the right script or was this script, which was never used before, used by accident or just that new?

Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:48]: We have two scripts that are similar: for release action and for validation action.
We can to think later is we need to optimize them to use one

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:9]:
```
~/java/github/romani/checkstyle [stoyanK7/issue/12488 L|✚ 1] 
$ git diff
diff --git a/.ci/releasenotes-gen-xdoc-push.sh b/.ci/releasenotes-gen-xdoc-push.sh
index 8056cd9..32856e3 100755
--- a/.ci/releasenotes-gen-xdoc-push.sh
+++ b/.ci/releasenotes-gen-xdoc-push.sh
@@ -2,11 +2,11 @@
 set -e
 
 source ./.ci/util.sh
-checkForVariable "READ_ONLY_TOKEN"
+#checkForVariable "READ_ONLY_TOKEN"
 
 if [[ -z $1 ]]; then
   echo "version is not set"
-  echo "Usage: update-releasenotesxml.sh <version>"
+  echo "Usage: $(basename $0) <version>"
   exit 1
 fi
 
✔ ~/java/github/romani/checkstyle [stoyanK7/issue/12488 L|✚ 1] 
$ .ci/releasenotes-gen-xdoc-push.sh
version is not set
Usage: releasenotes-gen-xdoc-push.sh <version>

```

Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh]: fixed

**Review by romani — APPROVED** [review]
ok to merge when CI pass

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (rnveach) [reply .ci/releasenotes-gen-xdoc-push.sh:49]: Then can we create an issue so its not forgotten?

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (stoyanK7) [reply .ci/releasenotes-gen-xdoc-push.sh]: Uh, realised that mistake just after commiting. Thanks for fixing it.

File: .ci/releasenotes-gen-xdoc-push.sh
Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:49]: I am not sure it makes sense. I waiting for final completion of auto release project, to rethink something at end. To clearly see what we need and what is convenient.

**Pull request merged**
